src/django/pfb_analysis/models.py

```diff
@@ -762,6 +770,22 @@ def _s3_url_for_result_resource(self, filename):
         )
 
 
+@receiver(post_delete, sender=AnalysisJob)
+def delete_analysisjob_s3_results(sender, instance, **kwargs):
+    s3 = boto3.resource('s3')
+    bucket = s3.Bucket(settings.AWS_STORAGE_BUCKET_NAME)
+    path = instance.s3_results_path
+    delete_result = bucket.objects.filter(Prefix=path).delete()
+    try:
+        logger.info("Deleted {} results files from {}".format(
+            sum(len(batch['Deleted']) for batch in delete_result),
```

Comment:
One tiny nitpick: if a neighborhood is deleted that has a queued job, the job path won't exist on S3 yet, but it'll try to delete it anyway. However, this doesn't seem to cause any errors, simply: `Deleted 0 results files from results/810d88c2-0b51-4118-a6df-45467dd76d54` .

I don't think this is worth addressing now but wanted to flag it so that the behavior is known.

Reply:
Yeah, true.  That seems fine to me, though.  The only ways I can think of to know whether there's anything to delete would be to infer from the job status or check the bucket.  The former seems fiddly and potentially brittle, and since the `delete()` call has no problem deleting zero items, adding an additional `list` call doesn't seem worth the trouble.